File: Go/bisectionmethod.go
package main

import (
	"fmt"
	"math"
)

func f(x float64) float64 {
	// return x*math.Exp(-x) + math.Pow(x, 3.0) + 1
	return (2/math.Exp(-math.Pi) + 1) * math.Exp(-x) * math.Sin(x)
}

func main() {
	n := 0
	eps := 1e-2
	var a float64
	var b float64
	fmt.Println("a: ")
	fmt.Scanln(&a)
	fmt.Println("b: ")
	fmt.Scanln(&b)

	fmt.Println("n", "a", "b", "m", "|Dx/2|")
	for {
		m := (a + b) / 2.0
		Dx := math.Abs((b - a) / 2.0)
		fmt.Println(n+1, a, b, m, Dx)
		if f(a)*f(m) < 0 {
			b = m
		} else {
			a = m
		}
		n++
		if Dx < eps {
			break
		}
	}
}

File: C#/BisectionMethod/bisectionmethod.cs
using System;

namespace BisectionMethod
{
	class MainClass
	{
		public static void Main (string[] args)
		{
			Console.Write ("a:");
			double a = Convert.ToDouble (Console.ReadLine ());
			Console.Write ("b:");
			double b = Convert.ToDouble (Console.ReadLine ());
			double eps = 1e-6;
			double m = 0;
			int i = 0;

			if (f(a) * f(m) > 0)
			{
				Console.Write ("No Root");
			} else
			{
				do {
					m = (a + b) / 2;
					Console.WriteLine ("{0},{1},{2},{3},{4},{5}", i + 1, a, b, m, f (a) * f (b) > 0 ? "+" : "-", Math.Abs ((b - a) / 2));

					if (f (a) * f (m) == 0) {

					}

					if (f (a) * f (m) < 0) {
						b = m;
					} else {
						a = m;
					}
					i++;
				
				} while (Math.Abs ((b - a) / 2) > eps);
			}
			Console.ReadLine ();
		}

		private static double f(double x)
		{
			return x * Math.Exp(-x) + Math.Pow(x, 3) + 1;
		}
	}
}
